oak_restricted_kernel/src/lib.rs

```diff
@@ -156,6 +156,36 @@ pub fn start_kernel(info: &BootParams) -> ! {
     // Safety: in the linker script we specify that the ELF header should be placed at 0x200000.
     let program_headers = unsafe { elf::get_phdrs(VirtAddr::new(0x20_0000)) };
 
+    let _stage0_dice_data = {
+        let dice_memory_slice = {
+            let e820_dice_data_entry = info
+                .e820_table()
+                .iter()
+                .find(|e| e.entry_type() == Some(oak_linux_boot_params::E820EntryType::DiceData))
+                .expect("failed to find dice data");
+
```

Comment:
initially I had another section here that validated that the E820 table matched any kernel args that are presented (nice to have, linux it them but restricted kernel could ignore the relevant kernel arg)

```
  // Validate that the dice data mem address matches the kernel args if present
  if let Some(kernel_arg_dice_addr) =
      kernel_args.get(oak_dice::evidence::DICE_DATA_CMDLINE_PARAM)
  {
      let parsed_addr =
          usize::from_str(kernel_arg_dice_addr).expect("failed to parse oak dice flag");
      if parsed_addr != e820_dice_data_entry.addr() {
          panic!("inconsistent dice data addresses supplied in the E820 and kernel args")
      }
  }
```

however, this led to memory issues, unsure why.